README.rst

```diff
@@ -0,0 +1,281 @@
+~~~~~~~~~~~~~~~~~~~~~
+
+To create client one should specify buffer's and network provider's
+implementations as template parameters. Connector's main class has the
+following signature:
+
+.. code-block:: c
+
+   template<class BUFFER, class NetProvider = DefaultNetProvider<BUFFER>>
+   class Connector;
+
+If one don't want to bother with implementing its own buffer or network
+provider, one can use default one: ``tnt::Buffer<16 * 1024>`` and
+``DefaultNetProvider<tnt::Buffer<16 * 1024>>``. So the default
+instantiation would look like:
+
+.. code-block:: c
+
+   using Buf_t = tnt::Buffer<16 * 1024>;
+   using Net_t = DefaultNetProvider<Buf_t >;
+   Connector<Buf_t, Net_t> client;
+
+Client itself is not enough to work with Tarantool instances, so let's
+also create connection objects. Connection takes buffer and network
+provider as template parameters as well (note that they must be the same
+as ones of client):
+
+.. code-block:: c
+
+   Connection<Buf_t, Net_t> conn(client);
+
+Connecting
+~~~~~~~~~~
+
+Now assume Tarantool instance is listening ``3301`` port on localhost.
+To connect to the server we should invoke ``Connector::connect()``
+method of client object and pass three arguments: connection instance,
+address and port.
+``int rc = client.connect(conn, address, port)``.
+
+Error handling
+~~~~~~~~~~~~~~
+
+Implementation of connector is exception free, so we rely on return
+codes: in case of fail, ``connect()`` will return ``rc < 0``. To get
+error message corresponding to the last error happened during
+communication with server, we can invoke ``Connection::getError()``
+method:
+
+.. code-block:: c
+
+   if (rc != 0) {
+       assert(conn.status.is_failed);
+       std::cerr << conn.getError() << std::endl;
+   }
+
+To reset connection after errors (clean up error message and connection
+status), one can use ``Connection::reset()``.
+
+Preparing requests
+~~~~~~~~~~~~~~~~~~
+
+To execute simplest request (i.e. ping), one can invoke corresponding
+method of connection object:
+``rid_t ping = conn.ping();``
+
+Each request method returns request id, which is sort of future. It
+can be used to get the result of request execution once it is ready
+(i.e. response). Requests are queued in the input buffer of connection
+until ``Connector::wait()`` is called.
+
+Sending requests
+~~~~~~~~~~~~~~~~
+
+That said, to send requests to the server side, we should invoke
+``client.wait()``:
+
+.. code-block:: c
+
+   client.wait(conn, ping, WAIT_TIMEOUT);
+
+Basically, ``wait()`` takes connection to poll (both IN and OUT),
+request id and optionally timeout (in milliseconds) parameters. once
+response for specified request is ready, ``wait()`` terminates. It also
+provides negative return code in case of system related fails (e.g.
+broken or time outed connection). If ``wait()`` returns 0, then response
+is received and expected to be parsed.
+
+Receiving responses
+~~~~~~~~~~~~~~~~~~~
+
+To get the response when it is ready, we can use
```

Comment:
In my opinion, it's best to consistently write "we can use", or "one can use", or just "use".